src/runtime/virtcontainers/container.go

```diff
@@ -639,6 +639,22 @@ func (c *Container) createBlockDevices(ctx context.Context) error {
 			c.mounts[i].Type = mntInfo.FsType
 			c.mounts[i].Options = mntInfo.Options
 			c.mounts[i].ReadOnly = readonly
+
+			for key, value := range mntInfo.Metadata {
+				switch key {
+				case volume.FSGroupMetadataKey:
+					gid, err := strconv.Atoi(value)
+					if err != nil {
+						c.Logger().WithError(err).Error(fmt.Sprintf("invalid group id provided for key %s", volume.FSGroupMetadataKey))
+						continue
+					}
+					c.mounts[i].FSGroup = uint32(gid)
+				case volume.FSGroupChangePolicyMetadataKey:
```

Comment:
Should we check the error case when `FSGroupMetadataKey` is missing and `FSGroupChangePolicyMetadataKey` exists?

Reply:
Thanks, added additional check when `FSGroupChangePolicyMetadataKey` exists but `FSGroupMetadataKey` is missing to log error and continue